packages/server/src/adapters/node-http/nodeHTTPRequestHandler.ts

```diff
@@ -31,36 +28,26 @@ export async function nodeHTTPRequestHandler<
       ? new URLSearchParams(opts.req.query as any)
       : new URLSearchParams(opts.req.url!.split('?')[1]);
 
-    const jsonContentTypeHandler =
-      defaultJSONContentTypeHandler as unknown as NodeHTTPContentTypeHandler<
-        TRequest,
-        TResponse
-      >;
-
-    const contentTypeHandlers = opts.experimental_contentTypeHandlers ?? [
-      jsonContentTypeHandler,
-    ];
-
-    const contentTypeHandler =
-      contentTypeHandlers.find((handler) =>
-        handler.isMatch({
-          ...opts,
-          query,
-        }),
-      ) ??
-      // fallback to json
-      jsonContentTypeHandler;
-
-    const bodyResult = await contentTypeHandler.getBody({
-      ...opts,
-      query,
-    });
+    // TODO: ensure this remains implemented at the right level
+    const preprocessedBody = false;
 
     const req: HTTPRequest = {
       method: opts.req.method!,
       headers: opts.req.headers,
       query,
-      body: bodyResult.ok ? bodyResult.data : undefined,
+      async getBodyJson() {
```

Comment:
Let's model HTTPRequest after an industry standard like the fetch API - .json .text .stream and return the appropriate types. Minimal risk of that being leaky for our needs